core/src/replicator.rs

```diff
@@ -122,13 +179,20 @@ impl Replicator {
     #[allow(clippy::new_ret_no_self)]
     pub fn new(
         ledger_path: &str,
-        node: Node,
+        mut node: Node,
         cluster_entrypoint: ContactInfo,
         keypair: Arc<Keypair>,
         _timeout: Option<Duration>,
     ) -> Result<Self> {
         let exit = Arc::new(AtomicBool::new(false));
 
+        // replicator cannot give information on rpc and
+        // cannot be leader so tpu/rpc ports are cleared
+        node.info.rpc = "0.0.0.0:0".parse().unwrap();
```

Comment:
maybe this is unnecessary. The node passed in should be setup for a replicator to begin with? Or take in a storage address and "make_replicator_node" like gossip_service does with spy_nodes?

Reply:
Sure, can do it that way.